src/kubernetes.cc

```diff
@@ -834,14 +848,26 @@ void KubernetesReader::WatchMaster(
           if (ec != boost::asio::error::operation_aborted) {
             watch_service->stop();
           }
+          });
+      }
+      std::unique_ptr<DelayTimer> staleness_timer;
+      if (config_.KubernetesUpdaterWatchReconnectMaxStalenessSeconds() > 0) {
+        staleness_timer = delay_timer_factory_->CreateTimer(*watch_service);
```

Comment:
I wonder if it makes sense to instead implement a periodic timer, either in the same class or a separate class? You're effectively implementing a `RunAsyncEvery` functionality. We could then refactor other periodic tasks, like `PollingMetadataUpdater`, to use the same functionality.

Reply:
Never mind, I've misread the code. See above.